crates/cli-support/src/webidl/mod.rs

```diff
@@ -623,6 +627,63 @@ impl<'a> Context<'a> {
         Ok(())
     }
 
+    fn discover_main(&mut self) -> Result<(), Error> {
+        let main_info = self
+            .module
+            .functions()
+            .find(|x| x.name.as_ref().map_or(false, |x| x == "main"))
+            .map(|x| (x.ty(), x.id()));
+        if let Some((main_type, main_id)) = main_info {
```

Comment:
Could this use a `match` with an early return to de-indent the rest of this body?

Reply:
it could indeed!